support-services/build.sbt

```diff
@@ -5,6 +5,8 @@ name := "support-services"
 description := "Scala library to provide shared services to Guardian Support projects."
 
 libraryDependencies ++= Seq(
+  "software.amazon.awssdk" % "dynamodb" % "2.13.26",
```

Comment:
the 1.x dynamo async client used a legacy way to contact dynamo which didn't allow the specification of a ConditionExpression together with attributes.  I needed to apply a condition to the update call to stop it from creating a redemption code if the provided one was not already in the DB(!!!)
there was a non async client that might have supported it, but I wouldn't want the whole server to hang if dynamo is being slow/timing out.
Aside from that, the 2.x client is supposed to be faster startup, more lightweight deps, and supports a nicer async (java 8) rather than the custom one.
On that basis we should endeavour to move everything across over time - it's fairly mind numbing though.

Reply:
https://mvnrepository.com/artifact/software.amazon.awssdk/dynamodb/2.13.26
```
$ cs fetch software.amazon.awssdk:dynamodb:2.13.26|xargs -I{} ls -sk {}|sed 's# .*/#kB #'|sort -n
kB annotations-2.13.26.jar
kB apache-client-2.13.26.jar
kB aws-core-2.13.26.jar
kB commons-logging-1.2.jar
kB http-client-spi-2.13.26.jar
kB httpclient-4.5.9.jar
kB jackson-annotations-2.10.4.jar
kB netty-codec-http-4.1.46.Final.jar
kB netty-common-4.1.46.Final.jar
kB sdk-core-2.13.26.jar
12kB reactive-streams-1.0.3.jar
24kB netty-reactive-streams-2.0.4.jar
28kB profiles-2.13.26.jar
32kB eventstream-1.0.1.jar
32kB protocol-core-2.13.26.jar
36kB netty-reactive-streams-http-2.0.4.jar
36kB netty-resolver-4.1.46.Final.jar
36kB netty-transport-native-unix-common-4.1.46.Final.jar
44kB slf4j-api-1.7.28.jar
104kB utils-2.13.26.jar
108kB aws-json-protocol-2.13.26.jar
128kB auth-2.13.26.jar
152kB netty-transport-native-epoll-4.1.46.Final-linux-x86_64.jar
236kB netty-nio-client-2.13.26.jar
272kB netty-buffer-4.1.46.Final.jar
316kB netty-codec-4.1.46.Final.jar
320kB httpcore-4.4.11.jar
328kB commons-codec-1.11.jar
344kB jackson-core-2.10.4.jar
408kB regions-2.13.26.jar
448kB netty-codec-http2-4.1.46.Final.jar
464kB netty-transport-4.1.46.Final.jar
492kB netty-handler-4.1.46.Final.jar
2096kB dynamodb-2.13.26.jar
2272kB jackson-databind-2.10.4.jar
```